### ISSUE TITLE
Apply "allowPlayerSchematics true" to player huts, not just decorations

### ISSUE THREAD

**Issue by Techiastronamo** [ldtteam/minecolonies#5426]
/title

Garfield227 told me to submit this regardless of what AshersLab  has to say about it, but I still don't get why this simple feature is purposefully left out "because we said so" when it's presumably already been implemented since it's claimed to be "too laggy" if that's truthfully why this isn't in the mod. Minecraft is about building so why try to get rid of that major component if you're trying to emulate Age of Empires? I know I'm not alone when I say that customization in AoE would be awesome so Minecraft is a perfect platform to allow that on top of all the other improvements Minecraft offers over the traditional AoE experience.

I already made a fork of this mod where I'll upload my own fix to this but I'm opening this in case someone wants to help with this implementing feature in the main branch of the mod. 

I know there is some mystical new fix coming out that addresses this issue "indirectly" or whatever they're saying in #mod-discussion, but I'm taking to mean that it probably doesn't actually do much in the ways of solving it because the explanation I've received is that it's being "kind of fixed as a side-effect" instead of a simple "yes" when I ask if it's going to solve my issue.

tl;dr- easy fix, if it's laggy just put a comment next to the option about it, or make a separate config option such as "allowPlayerHutSchematics true" with that comment if you want to allow custom player decorations but not custom player huts.

Thanks!

**Comment by Raycoms** [comment]
Oh, it's not about it being laggy. It's about making a coherent play experience on servers. When future parts like economics and combat come into play and you make your colony a bunch of dirt houses without any additional requirements it will give you a huge edge over everyone else. This way there is no challenge left to solve. One of the core challenges of the mods is to setup a self sustainable colony. If all your houses are only made from dirt and cobble then that becomes surprisingly easy very quickly.

So yes, this is on purpose. In singleplayer you can build your own ones already and use them and in multiplayer they have to be uploaded to the server to allow using them.

**Comment by Techiastronamo** [comment]
But the entire point is to eliminate the file management on the server side, who are you to say what is and isn't an acceptable building? We want ease of customization, which presently is purposefully being hindered by this hard stance against it.

**Comment by Raycoms** [comment]
1) we're the coders who know what the workers need to have to be able to
work, I don't want 1000 bug reports about my worker not working because
people messed up the custom builds
2) like I said, we give things a minimum complexity to not make it too
easy. If you don't want this you can do what you want on your server. But
I'm not actively going to break the game model we've created. To go back to
age of empires, the cost of the buildings should still be similar even when
customazible, else it breaks the entire game when buildings cost only 1
wood

**Comment by Techiastronamo** [comment]
Ok doing it your way- "In singleplayer you can build your own ones already and use them and in multiplayer they have to be uploaded to the server to allow using them." We tested this, it does not work. We went step-by-step in the wiki and the custom hut schematics were unusable in both singleplayer AND multiplayer. Instead of showing up in the hut building UI, it shows up as "MySchematics" where the hut never actually places but the rest of the construction does get placed.

The only solution so far that I've found is to insert the blueprints into the jar itself, but that would require recompiling and redistributing the entire mod.

**Comment by Raycoms** [comment]
That works, you call the schematic the same way it is in the jar and on the wiki, you put the hut block into it and you put it in the right folder structurize/schematics/stylename. I test this regularly and it works fine

**Comment by Techiastronamo** [comment]
There's our problem then, we put it in Minecolonies instead of Structurize, despite both having scans and schematics folders. Thanks.
